src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Data/ToggleBetweenFeatureRequestModes/ToggleBetweenFeatureRequestModes.xaml.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright 2022 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+// language governing permissions and limitations under the License.
+
+using Esri.ArcGISRuntime.Data;
+using Esri.ArcGISRuntime.Mapping;
+using System;
+using System.Windows;
+
+namespace ArcGISRuntime.WPF.Samples.ToggleBetweenFeatureRequestModes
+{
+    [ArcGISRuntime.Samples.Shared.Attributes.Sample(
+        name: "Toggle between feature request modes",
+        category: "Data",
+        description: "Use different feature request modes to populate the map from a service feature table.",
+        instructions: "Run the sample and use the radio buttons to change what feature request modes you want to use (the default value is  **on interaction cache**). After you selected which feature request mode to use, click the `Populate` button to apply the feature request mode. ",
+        tags: new[] { "cache", "feature request mode", "performance" })]
+    public partial class ToggleBetweenFeatureRequestModes
+    {
+        private ServiceFeatureTable _treeFeatureTable;
+        private FeatureLayer _treeFeatureLayer;
+
+        public ToggleBetweenFeatureRequestModes()
+        {
+            InitializeComponent();
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            try
+            {
+                // Create new Map with basemap.
+                Map myMap = new Map(BasemapStyle.ArcGISTopographic);
+                MyMapView.Map = myMap;
+
+                // Set intial map location.
+                MyMapView.Map.InitialViewpoint = new Viewpoint(45.5266, -122.6219, 6000);
+
+                // Create uri to the used feature service.
+                Uri serviceUri = new Uri(
+                   "https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/rest/services/Trees_of_Portland/FeatureServer/0");
+
+                // Create a feature table for the trees of Portland feature service.
+                // The feature request mode for this service feature table is OnInteractionCache by default.
+                _treeFeatureTable = new ServiceFeatureTable(serviceUri);
+
+                // Create FeatureLayer that uses the created table.
+                _treeFeatureLayer = new FeatureLayer(_treeFeatureTable);
+
+                // Add created layer to the map.
+                MyMapView.Map.OperationalLayers.Add(_treeFeatureLayer);
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show(ex.ToString(), "Error");
```

Comment:
```suggestion
                MessageBox.Show(ex.Message, ex.GetType().Name, MessageBoxButton.OK, MessageBoxImage.Error);
```
`ex.ToString()` doesnt give very good information. I recommend using ex.message in the error messages for this implementation.